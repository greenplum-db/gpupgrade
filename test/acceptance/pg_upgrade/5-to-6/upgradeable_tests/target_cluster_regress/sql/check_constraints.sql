-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------

SELECT * FROM heap_table_with_check_constraint;
-- this insert should fail
INSERT INTO heap_table_with_check_constraint VALUES (2, 'Jane');

SELECT * FROM partition_table_with_check_constraint;
-- this insert should fail
INSERT INTO partition_table_with_check_constraint VALUES (1, 1, 3);

