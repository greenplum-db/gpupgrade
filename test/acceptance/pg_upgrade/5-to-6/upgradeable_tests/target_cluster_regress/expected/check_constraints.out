-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------

SELECT * FROM heap_table_with_check_constraint;
 id | name 
----+------
 1  | Joe  
(1 row)
-- this insert should fail
INSERT INTO heap_table_with_check_constraint VALUES (2, 'Jane');
ERROR:  new row for relation "heap_table_with_check_constraint" violates check constraint "heap_table_with_check_constraint_id_check"
DETAIL:  Failing row contains (2, Jane).

SELECT * FROM partition_table_with_check_constraint;
 a  | b  | c  
----+----+----
 1  | 1  | 1  
 2  | 2  | 2  
 3  | 3  | 3  
 4  | 4  | 4  
 5  | 5  | 5  
 6  | 6  | 6  
 7  | 7  | 7  
 8  | 8  | 8  
 9  | 9  | 9  
 10 | 10 | 10 
(10 rows)
-- this insert should fail
INSERT INTO partition_table_with_check_constraint VALUES (1, 1, 3);
ERROR:  new row for relation "partition_table_with_check_constraint_1_prt_part" violates check constraint "a_check"
DETAIL:  Failing row contains (1, 1, 3).

